packages/components/space/src/Space.vue

```diff
@@ -0,0 +1,98 @@
+      <div class="ix-space-item" :class="child.className" :style="child.style">
+        <component :is="child.node" />
+      </div>
+      <slot v-if="index < spaceList.length - 1" name="split">
+        <span v-if="split" :key="child.id + 'span'" class="ix-space-split">
+          {{ split }}
+        </span>
+      </slot>
+    </template>
+  </div>
+</template>
+
+<script lang="ts">
+import type { SetupContext, VNode } from 'vue'
+import type { SpaceProps } from './types'
+
+import { computed, defineComponent } from 'vue'
+import { PropTypes } from '@idux/cdk/utils'
+import { Logger } from '@idux/components/core/logger'
+import { getSlotNodeList } from '@idux/components/core/utils'
+import { useGlobalConfig, SpaceConfig } from '@idux/components/core/config'
+
+interface SpaceItem {
+  style?: Partial<CSSStyleDeclaration>
+  className?: string
+  id: number
+  node: VNode
+}
+
+const spaceSizePropType = PropTypes.oneOf(['small', 'medium', 'large'] as const)
+export default defineComponent({
+  name: 'IxSpace',
+  props: {
+    align: PropTypes.oneOf(['start', 'center', 'end', 'baseline'] as const).def('baseline'),
+    direction: PropTypes.oneOf(['vertical', 'horizontal'] as const).def('horizontal'),
+    size: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.oneOfType([spaceSizePropType, PropTypes.number])),
+      PropTypes.oneOfType([spaceSizePropType, PropTypes.number]),
+    ]),
+    split: PropTypes.string,
+    wrap: PropTypes.bool,
+  },
+  setup(props: SpaceProps, { slots }: SetupContext) {
+    const spaceConfig = useGlobalConfig('space')
+    const children = getSlotNodeList(slots)
+    const spaceList = useSpaceList(props, spaceConfig, children, !!slots.split)
+
+    const className = useClassName(props)
+
+    return { spaceList, className }
+  },
+})
+
+function useSpaceList(props: SpaceProps, spaceConfig: SpaceConfig, children: VNode[], withSplit: boolean) {
+  return computed(() => {
+    const size = props.size ?? spaceConfig.size
+    const split = !!props.split || withSplit
+
+    let spaceList: SpaceItem[]
+    if (size instanceof Array) {
+      if (children.length - 1 !== size.length) {
+        Logger.warn('The number of split elements is inconsistent with the length of the size array')
+      }
+      spaceList = children.map((child, index) => {
+        const current: SpaceItem = { node: child, id: index }
+        if (split) return current
+        else if (typeof size[index] === 'number') {
+          current.style = { marginRight: `${size[index]}px` }
+        } else current.className = `ix-space-item-${size[index]}`
+        return current
+      })
+    } else {
+      spaceList = children.map((child, index) => {
+        const current: SpaceItem = { node: child, id: index }
+        if (split) return current
+        else if (typeof size === 'number') {
+          current.style = { marginRight: `${size}px` }
+        } else current.className = `ix-space-item-${size}`
```

Comment:
```suggestion
        } else {
        current.className = `ix-space-item-${size}`
        }
```
另外在做 style 和 className 赋值的时候，需不需要考虑用户本身节点上的 style 和 className？ 你这样做会不会直接把用户自己设置的给干掉了。

Reply:
> 另外在做 style 和 className 赋值的时候，需不需要考虑用户本身节点上的 style 和 className？ 你这样做会不会直接把用户自己设置的给干掉了。

不会，因为这个赋值的本身是组件内部的元素节点，不会影响到调用方传进来的模板